### PR TITLE
allow loading custom word2vec models(including model saved as binary file)

### PR DESCRIPTION
**Pull request by ashutoshsingh0223**
Take custom path to a word2vec model saved on disk and load it through gensim.

In case model has been saved as a binary file load it through gensim.models.KeyedVectors.load_word2vec_format with "binary"  argument set to True

### CODE REVIEW

File: flair/embeddings.py [hunk 1]
```diff
@@ -164,10 +164,12 @@ def __str__(self):
 class WordEmbeddings(TokenEmbeddings):
     """Standard static word embeddings, such as GloVe or FastText."""
 
-    def __init__(self, embeddings: str, field: str = None):
+    def __init__(self, embeddings: str, field: str = None, path: str = None, load_binary: bool = False):
         """
         Initializes classic word embeddings. Constructor downloads required files if not there.
         :param embeddings: one of: 'glove', 'extvec', 'crawl' or two-letter language code.
+        :param path(defaults to None): path of custom word2vec model (can also be a binary file)
```
[the remark this answers is not in the document]
Comment (ashutoshsingh0223) [reply flair/embeddings.py]: Yes...`path` can be removed. Will do that.
But there is no indicator for if the model is binary... `load_binary` serves that purpose.
Plus it makes usage of `WordEmbeddings` class more inline with `gensim` where you set `binary = True` if you want to load model from a binary file.

File: flair/embeddings.py [hunk 2]
```diff
@@ -274,16 +276,28 @@ def __init__(self, embeddings: str, field: str = None):
...
                 f'The given embeddings "{embeddings}" is not available or is not a valid path.'
             )
 
         self.name: str = str(embeddings)
         self.static_embeddings = True
 
-        self.precomputed_word_embeddings = gensim.models.KeyedVectors.load(
-            str(embeddings)
-        )
+        if load_binary:
```
[the remark this answers is not in the document]
Comment (ashutoshsingh0223) [reply flair/embeddings.py]: I'll check and get back very soon

[the remark this answers is not in the document]
Comment (ashutoshsingh0223) [reply flair/embeddings.py]: We can directly check for `.bin` extention.

or use `mimetypes`(inbuilt in python3)
```
import mimetypes
binary = False
if ".bin" in mimetypes.guess_all_extensions(
                                          mimetypes.guess_type(path,strict=True)[0],
                                          strict=True):
    binary = True
```


I think the first method would be enough as `gensim` saves binary files with `bin` extention

[the remark this answers is not in the document]
Comment (ashutoshsingh0223) [reply flair/embeddings.py]: Or leave it like this. It really feels closer to the way `gensim` does it which very popularly used

[the remark this answers is not in the document]
Comment (stefan-it) [reply flair/embeddings.py]: I think a simple file extension check (`.endswith('.bin')`) should be sufficient here :)

[the remark this answers is not in the document]
Comment (ashutoshsingh0223) [reply flair/embeddings.py]: Cool

**Comment by alanakbik** [comment]
Great, thanks!

**Comment by alanakbik** [comment]
:+1:

**Comment by ashutoshsingh0223** [comment]
@alanakbik Need two approvals.

**Comment by ashutoshsingh0223** [comment]
Can you help me with that??

**Comment by ashutoshsingh0223** [comment]
@stefan-it Can you approve??

**Comment by stefan-it** [comment]
It's currently not working for me:

```python
import flair
from flair.embeddings import WordEmbeddings
embd = WordEmbeddings('wiki.fa.bin')
```

Error message:

```bash
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/flair/flair/embeddings.py", line 286, in __init__
    str(embeddings), binary=True
  File "/media/stefan/00eb2e8f-6826-4043-a1d7-937b9d5341c1/.venvs/flair/lib/python3.7/site-packages/gensim/models/keyedvectors.py", line 1476, in load_word2vec_format
    limit=limit, datatype=datatype)
  File "/media/stefan/00eb2e8f-6826-4043-a1d7-937b9d5341c1/.venvs/flair/lib/python3.7/site-packages/gensim/models/utils_any2vec.py", line 343, in _load_word2vec_format
    header = utils.to_unicode(fin.readline(), encoding=encoding)
  File "/media/stefan/00eb2e8f-6826-4043-a1d7-937b9d5341c1/.venvs/flair/lib/python3.7/site-packages/gensim/utils.py", line 359, in any2unicode
    return unicode(text, encoding, errors=errors)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xba in position 0: invalid start byte
```

🤔

**Comment by ashutoshsingh0223** [comment]
Can you try for some other file??

**Comment by ashutoshsingh0223** [comment]
Also how did you train the custom model??

**Comment by stefan-it** [comment]
I was using the Persian pretrained FastText embeddings from [here](https://github.com/facebookresearch/fastText/blob/master/docs/pretrained-vectors.md) - but I'm also currently testing some other pretrained embeddings and will report back :)

**Comment by ashutoshsingh0223** [comment]
`load_word2vec_format` is not meant for loading `fasttext`

**Comment by stefan-it** [comment]
Ah ok.

Normally, I would load a FastText model with:

```python
import gensim

language = "fa"
word_vectors = gensim.models.KeyedVectors.load_word2vec_format(f'wiki.{language}.vec', binary=False)
word_vectors.save(f'wiki.{language}.vec.gensim')

embeddings = WordEmbeddings(f'wiki.{language}.vec.gensim')
```

so I think this should also be integrated (in maybe a follow-up) PR :)

**Comment by stefan-it** [comment]
Anyway, I tried the current PR with the German Word Embeddings from [here](https://devmount.github.io/GermanWordEmbeddings/) and the model could be imported :)

**Comment by stefan-it** [comment]
:+1:

**Comment by alanakbik** [comment]
@stefan-it thanks for checking this!

**Pull request merged**